src/Symfony/Bundle/FrameworkBundle/HttpClient/TraceableHttpClient.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\HttpClient;
+
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+use Symfony\Contracts\HttpClient\ResponseStreamInterface;
+
+/**
+ * @author Jérémy Romey <jeremy@free-agent.fr>
+ */
+class TraceableHttpClient implements HttpClientInterface
+{
+    protected $httpClient;
+    protected $traces = [];
+
+    /**
+     * @param HttpClientInterface $httpClient A HttpClientInterface instance
+     */
+    public function __construct(HttpClientInterface $httpClient)
+    {
+        $this->httpClient = $httpClient;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function request(string $method, string $url, array $options = []): ResponseInterface
+    {
+        $response = $this->httpClient->request($method, $url, $options);
+
+        $this->traces[] = [
+            'request' => [
+                'method' => $method,
+                'url' => $url,
+                'options' => $options,
+            ],
+            'response' => [
+                'statusCode' => $response->getStatusCode(),
```

Comment:
This broke the laziness of responses and avoid the response to be handled in an async way. 

See https://github.com/symfony/symfony/pull/30413#issuecomment-469902643